bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpClientFacade.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http;
+
+/**
+ * A facade to the jetty http client
+ * 
+ * @author Michael Bock - initial API
+ */
+import java.net.CookieStore;
+import java.net.URI;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+import org.eclipse.jetty.client.api.Connection;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Destination;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.util.Fields;
+
+/**
+ * A facade to Jetty {@link HttpClient}
+ * 
+ * @author Michael Bock - initial API
+ */
+public interface HttpClientFacade {
```

Comment:
I would simply call it `HttpClient`. The fact that it is a facade is nothing that needs to be exposed to the user. It would simply be the `HttpClient` in the `org.eclipse.smarthome.io.net.http` package.

Reply:
The other method in the factory returns a plain jetty client. If we use the same name, the factory would return two different HttpClient classes. Isn't that a little bit confusing?

Reply:
As clients will normally either use the one or the other, I think this would be ok. If you think this is an issue, I'd maybe go for a name like `ESHHttpClient` or `CommonHttpClient`.